lms/djangoapps/support/tests/test_views.py

```diff
@@ -1369,3 +1381,211 @@ def test_linking_program_enrollment_without_username(self, username, original_us
         response_data = json.loads(response.content.decode('utf-8'))
         error = "All linking lines must be in the format 'external_user_key,lms_username'"
         assert response_data['errors'] == [error]
+
+
+class TestOnboardingView(SupportViewTestCase, ProctoredExamTestCase):
+    """
+    Tests for OnboardingView
+    """
+
+    def setUp(self):
+        super().setUp()
+        SupportStaffRole().add_users(self.user)
+
+        self.proctored_exam_id = self._create_proctored_exam()
+        self.onboarding_exam_id = self._create_onboarding_exam()
+
+        self.other_user = User.objects.create(username='otheruser', password='test')
+        self.other_course_content = 'block-v1:test+course+2+type@sequential+block@other_onboard'
+
+        self.other_course = CourseFactory.create(
+            org='x',
+            course='y',
+            run='z',
+            enable_proctored_exams=True,
+            proctoring_provider=settings.PROCTORING_BACKENDS['DEFAULT'],
+        )
+
+        yesterday = timezone.now() - timezone.timedelta(days=1)
+        self.course_scheduled_sections = {
+            BlockUsageLocator.from_string(self.content_id_onboarding): MockScheduleItemData(yesterday),
+            BlockUsageLocator.from_string(self.other_course_content): MockScheduleItemData(yesterday),
+        }
+
+        set_runtime_service('learning_sequences', MockLearningSequencesService(
+            list(self.course_scheduled_sections.keys()),
+            self.course_scheduled_sections,
+        ))
+
+        self.onboarding_exam = ProctoredExam.objects.get(id=self.onboarding_exam_id)
+
+    def tearDown(self):  # lint-amnesty, pylint: disable=super-method-not-called
+        """
+        Do not call super method as it will delete default event trackers.
+        tearDownClass is sufficient in this case.
+        """
+        return
```

Comment:
why is this needed if it is not doing anything? The docstring is unclear.

Reply:
+1 to this as well. Haven't seen many instances like this one before.

Reply:
+1

Reply:
This is to **override** default method which requires and calls `super().tearDown()` to run inherited `tearDown()` methods. We have two parent classes here, `SupportViewTestCase` and `ProctoredExamTestCase`. `SupportViewTestCase` has no `tearDown()` method hence we are fine but `ProctoredExamTestCase` has an inherited `tearDown()` method from [here](https://github.com/openedx/edx-proctoring/blob/211ca866ae79e8f99ebab091a7a1abbd4d4ff791/edx_proctoring/tests/utils.py#L206) which delete default key from `TRACKERS` leading to failure of all subsequent tests.

Reply:
updated docstring as well.